app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt

```diff
@@ -4,19 +4,40 @@
 package org.mozilla.fenix.components.metrics
 
 import android.content.Context
+import mozilla.components.service.glean.EventMetricType
 import mozilla.components.service.glean.Glean
+import mozilla.components.support.utils.Browsers
 import org.mozilla.fenix.BuildConfig
 import org.mozilla.fenix.utils.Settings
+import org.mozilla.fenix.debug.GleanMetrics.Metrics
+import org.mozilla.fenix.debug.GleanMetrics.Events
+
+private val Event.metricType: EventMetricType?
+    get() = when (this) {
+        is Event.OpenedApp -> Events.appOpened
+        is Event.SearchBarTapped -> Events.searchBarTapped
+        is Event.EnteredUrl -> Events.enteredUrl
+        is Event.PerformedSearch -> Events.performedSearch
+        else -> null
+    }
 
 class GleanMetricsService(private val context: Context) : MetricsService {
     override fun start() {
         Glean.initialize(context)
         Glean.setUploadEnabled(IsGleanEnabled)
+
+        Metrics.apply {
+            defaultBrowser.set(Browsers.all(context).isDefaultBrowser)
+        }
     }
 
-    override fun track(event: Event) { }
+    override fun track(event: Event) {
+        event.metricType?.record(event.extras)
+    }
 
-    override fun shouldTrack(event: Event): Boolean = Settings.getInstance(context).isTelemetryEnabled
+    override fun shouldTrack(event: Event): Boolean {
```

Comment:
Glean does this internally based on a persistent internal flag that can be toggled using the [`Glean.setUploadEnabled()`](https://mozac.org/api/mozilla.components.service.glean/-glean-internal-a-p-i/set-upload-enabled.html) function.  The idea was to toggle the flag when the user opted out of telemetry and let the glean library handle discarding recorded data and preventing upload.

Reply:
@travis79 Good to know, the secondary case for this method is we need a way to allow/block some metrics from being sent to some providers to support Leanplum

Reply:
Then that's a good case for doing this in the client app.  Glean currently doesn't have a way to disable a single specific metric at run-time, only through the metrics.yaml file.

Reply:
It probably wouldn't be too hard to make that work -- but maybe it doesn't buy much...

Reply:
@travis79 @mdboom If you look at the code in `metrics.kt` you can see how I have it setup. Right now I have a generic `Event` type for things that I want to track inside Fenix. When we track an event it dispatches each event to a `MetricsService` which decides whether or not it wants to track that event and then transforms is appropriately into a format that service expects.